### PR TITLE
feat(swc/node): interop swc/wasm for bindings

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

- closes https://github.com/swc-project/swc/issues/5221

PR includes 2 changes:

1. Introduce `postinstall` to silently install `@swc/wasm` if native binding cannot be loaded
There's no good way to specify fallback dependencies if all of optional dependency installation fails. A naive postinstall script does validation of installation, then installs `@swc/wasm` as fallback if native binary cannot be loaded.

To reduce burden of installation pkg for the features only will be used for rare cases of fallback, postinstall script doesn't try to use own dependencies to resolve & download pkg but relies on system's npm directly. 

2. Binding falls back to `@swc/wasm`

In actual bindings, if native binding is not available it'll try to import `@swc/wasm` and interop its interface if @swc/wasm have corresponding one. If there aren't, it'll transparently fail.

PR doesn't attempt to fill the gaps between `@swc/wasm` and native bindings interface. That's something we may want to improve over, but probably not a high priority.

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - ''

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Comment by kwonoj** [comment]
CI's now good to go, though this is a bit tricky to test, so we may need to see a couple of releases to see if there are some regressions.

**Review by kdy1 — APPROVED** [review]

**Comment by sinceresiva** [comment]
Hi,
Let me give the background for reference. Grafana v9.0.3 is using a previous version of @swc/core which was giving the "Unsupported Architecture" error in IBM z-Linux (s390x) machines. I have now changed the Grafana package.json to use the latest version of @swc/core **v1.2.218**. Now I get a different error "Bindings not found" as show in the error in below screenshot (on executing: yarn start). From the current PR **swc** is supposed to interop between @swc/core and @swc/wasm. Am I missing any additional step? Pls help.
image
My environment is:
OS: IBM z-Linux (s390x)
NodeJS version: v16.15.1
yarn version: 4.0.0-rc.12
npm version: 8.11.0

File: node-swc/src/postinstall.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  const installDir = path.join(coreDir, 'npm-install');
+
+  try {
+    fs.mkdirSync(installDir);
+    fs.writeFileSync(path.join(installDir, 'package.json'), '{}');
+
+    // Instead of carrying over own dependencies to download & resolve package which increases installation sizes of `@swc/core`,
+    // assume & relies on system's npm installation.
+    child_process.execSync(`npm install --no-save --loglevel=error --prefer-offline --no-audit --progress=false @swc/wasm@${version}`,
+      { cwd: installDir, stdio: 'pipe', env });
```
[the remark this answers is not in the document]
Comment (kwonoj) [reply node-swc/src/postinstall.ts:99]: This is the way esbuild does for fallback as well : https://github.com/evanw/esbuild/blob/54ebf0cc7f7580e9b290a5a225ae8c784745d45f/lib/npm/node-install.ts#L97

**Pull request merged**